dart/constraint/ConstraintSolver.cpp

```diff
@@ -478,6 +479,29 @@ void ConstraintSolver::updateConstraints()
   // Destroy previous soft contact constraints
   mSoftContactConstraints.clear();
 
+  // Create a mapping of contact pairs to the number of contacts between them
+  using ContactPair
+      = std::pair<collision::CollisionObject*, collision::CollisionObject*>;
+
+  // Compare contact pairs while ignoring their order in the pair.
+  struct ContactPairCompare
+  {
+    ContactPair SortedPair(const ContactPair& a) const
+    {
+      if (a.first < a.second)
+        return std::make_pair(a.second, a.first);
+      return a;
+    }
+
+    bool operator()(const ContactPair& a, const ContactPair& b) const
+    {
+      // Sort each pair and then do a lexicographical comparison
+      return SortedPair(a) < SortedPair(b);
+    }
+  };
+
+  std::map<ContactPair, size_t, ContactPairCompare> contactPairMap;
```

Comment:
Consider using `std::unordered_map` as the order of the pair doesn't matter in the use cases

Reply:
I opted for using a `std::map` because I thought a writing a compare function was easier than writing a hash function. If you think performance would be a problem, I can look into a hash function for `ContactPair` and using `std::unordered_map`.